packages/perseus-editor/src/components/device-framer.tsx

```diff
@@ -24,67 +25,72 @@ const SCREEN_SIZES = {
     },
 } as const;
 
-type Props = any;
-
-class DeviceFramer extends React.Component<Props> {
-    static propTypes = {
-        children: PropTypes.element.isRequired,
-        deviceType: PropTypes.oneOf([
-            constants.devices.PHONE,
-            constants.devices.TABLET,
-            constants.devices.DESKTOP,
-        ]).isRequired,
-        // TODO(kevinb) rename to variableHeight
-        nochrome: PropTypes.bool,
-    };
-
-    render(): React.ReactNode {
-        const deviceType = this.props.deviceType;
+type Props = React.PropsWithChildren<{
+    deviceType?: DeviceType;
+    nochrome: boolean;
+}>;
 
-        if (this.props.nochrome) {
-            // Render content inside a variable height iframe.  Used on the
-            // "edit" table of the content editor. In this mode, PerseusFrame
-            // will draw the border and reserve space on the right for
-            // lint indicators.
-            return (
-                <div>
-                    <div
-                        key="screen"
-                        style={{
-                            width:
-                                SCREEN_SIZES[deviceType].framedWidth +
-                                2 * constants.perseusFrameBorderWidth +
-                                constants.lintGutterWidth,
-                        }}
-                    >
-                        <div>{this.props.children}</div>
-                    </div>
-                </div>
-            );
-        }
-        const scale =
+const DeviceFramer = ({
+    children,
+    deviceType = "phone",
+    nochrome,
+}: Props): React.ReactElement => {
+    const scale = React.useMemo(
+        () =>
             SCREEN_SIZES[deviceType].framedWidth /
-            SCREEN_SIZES[deviceType].width;
+            SCREEN_SIZES[deviceType].width,
+        [deviceType],
+    );
 
-        // In this mode we draw our own border and don't reserve
-        // space for a lint gutter.
-        const screenStyle = {
+    // In this mode we draw our own border and don't reserve
+    // space for a lint gutter.
+    const withChromeStyle = React.useMemo(
+        () => ({
             backgroundColor: "white",
+            overflow: "scroll",
             color: "black",
             textAlign: "left",
             width: SCREEN_SIZES[deviceType].width,
             height: SCREEN_SIZES[deviceType].height,
             border: "solid 1px #CCC",
             margin: 8,
             zoom: scale,
-        } as const;
+        }),
+        [deviceType, scale],
+    );
 
+    if (nochrome) {
```

Comment:
The rendering part of this component required some re-organization to migrate it because of the conditional rendering.